File: app/src/main/java/com/clug/nanal/activity/Survey4_3Activity.java
package com.clug.nanal;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;

public class Survey4_3Activity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_survey4_3);
    }

    public void goToSurvey5(View v){
        Intent intent=new Intent(this, Survey5Activity.class);
        startActivity(intent);
    }

    int getUserTemp(int userTemp, int temp){
        CheckBox pants=(CheckBox)findViewById(R.id.pants);
        CheckBox thickPants=(CheckBox)findViewById(R.id.thick_pants);
        CheckBox stocking=(CheckBox)findViewById(R.id.stocking);
        CheckBox leggings=(CheckBox)findViewById(R.id.leggings);
        userTemp=temp;

        if(pants.isChecked()) userTemp+=1;
        if(leggings.isChecked()) userTemp+=2;
        if(thickPants.isChecked()) userTemp+=3;
        if(stocking.isChecked()) userTemp+=1;

        return userTemp;
    }
}

File: app/src/main/java/com/clug/nanal/activity/Survey4_3Activity.kt
package com.clug.nanal.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.CheckBox
import android.widget.EditText
import android.widget.TextView
import com.clug.nanal.R
import kotlinx.android.synthetic.main.activity_survey2.*
import kotlinx.android.synthetic.main.activity_survey3.*
import kotlinx.android.synthetic.main.activity_survey4_1.*
import kotlinx.android.synthetic.main.activity_survey4_2.*
import kotlinx.android.synthetic.main.activity_survey4_3.*

class Survey4_3Activity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_survey4_3)

        val preTempo = intent.extras!!.getFloat("tempo")

        btn_survey4_3_ok.setOnClickListener {
            val intent = Intent(this, Survey5Activity::class.java)
            intent.putExtra("tempo", getUserTemp(preTempo))
            startActivity(intent)
            finish()
        }
    }

    internal fun getUserTemp(temp: Float): Float {
        var userTemp = temp

        if (pants.isChecked) userTemp += 1
        if (leggings.isChecked) userTemp += 2
        if (thick_pants.isChecked) userTemp += 3
        if (stocking.isChecked) userTemp += 1

        return userTemp
    }
}